nautobot/extras/api/views.py

```diff
@@ -80,18 +80,37 @@ def get_view_name(self):
 
 
 class NotesViewSetMixin:
+    def restrict_queryset(self, request, *args, **kwargs):
+        """
+        Apply "view" permissions on the POST /notes/ endpoint, otherwise as ModelViewSetMixin.
+        """
+        if request.user.is_authenticated and self.action == "notes":
+            self.queryset = self.queryset.restrict(request.user, "view")
+        else:
+            super().restrict_queryset(request, *args, **kwargs)
+
+    class CreateNotePermissions(TokenPermissions):
+        """As nautobot.core.api.authentication.TokenPermissions, but enforcing add_note permission."""
+
+        perms_map = {
+            "GET": ["%(app_label)s.view_%(model_name)s", "extras.view_note"],
+            "POST": ["%(app_label)s.view_%(model_name)s", "extras.add_note"],
```

Comment:
Do we need `"%(app_label)s.view_%(model_name)s"` for the `POST` action here? (Do we need permission to view the model in order to add notes to it?)

Reply:
I was thinking so - you shouldn't be able to add notes to an object that you can't see? Open to discussion on this point as this was just my best guess as to how it "should" work.